### ISSUE TITLE
Remove mentor mode button in header for judges with only judge account when mentor registration is closed

### ISSUE THREAD

**Issue by dboyer** [Iridescent-CM/technovation-app#4629]
Currently, when judges register, they automatically have a mentor mode button in the header that lets them view a mentor account, even if they only have a judge profile and mentor registration is closed. We should not display this button if the person only has a judge account and mentor registration is closed.

If the person already has a mentor account (i.e. they are a mentor who also judges), the mentor mode button should remain visible at all times so they can switch back to their mentor profile.

For judges, hide the "Switch to Mentor Mode" button, if:

- ENABLE_SWITCH_BETWEEN_JUDGE_AND_MENTOR is set to true OR
- mentor registration is open OR
- mentor registration is closed AND the judge already has a mentor profile

**Comment by dboyer** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Technovation-605271851956b1000ec2f24d/issues/Iridescent-CM/technovation-app/4629?planning-poker) with Zenhub @shaun-technovation @viviancan

**Comment by shaun-technovation** [comment]
Can I double check/verify the logic... only display the mentor switch button if:

- ENABLE_SWITCH_BETWEEN_JUDGE_AND_MENTOR is set to true OR
- mentor registration is open (for judges who have a mentor profile or not) OR
- mentor registration is closed AND they have a mentor profile

Is that correct? 🤔

**Comment by shaun-technovation** [comment]
Just documenting from our standup that this is how the mentor switch button will work for judges, it will be enabled if:

- ENABLE_SWITCH_BETWEEN_JUDGE_AND_MENTOR is set to true OR
- mentor registration is open OR
- mentor registration is closed AND the judge already has a mentor profile

**Comment by shaun-technovation** [comment]
This is on QA.

**Comment by dboyer** [comment]
Tested on QA:

- Mentor registration was off. I removed the config variable. I logged in as a judge with just a judge profile and did not have the mentor mode button.
- Mentor registration was off. I removed the config variable. I logged in as a judge who also had a mentor profile and the mentor mode button was visible.
- Mentor registration was off. I removed the config variable. I logged in as a judge who also had a mentor profile and the mentor mode button was visible. I clicked it to go to the mentor dashboard but **the button to go to my judge profile was gone so I was not able to easily start judging**.
- Mentor registration was off. I added the config variable. I logged in as a judge with just a judge profile and saw the mentor mode button. **When I clicked it, I got the error page**.
- Mentor registration was on. I added the config variable. I logged in as a new judge without a mentor profile and saw the mentor mode button. I was able to click on it and go to the mentor dashboard.

I think the issue is with the config button. If I remove it, everything works fine for judges but mentors can't get to their judge profile. If I keep it, judges still see the mentor mode button when they shouldn't. 

Maybe we need to escalate #4679

**Comment by shaun-technovation** [comment]
Here's an idea...

Two new configuration settings for judges:
- Enable mentor mode for judges who already have a mentor profile
- Enable mentor mode for judges who don't currently have a mentor profile

And one new configuration setting for mentors (or maybe it could be the same as above):
- Enable judge mode for all mentors

----

And, when we convert this to the admin area, it could be something like this (we could also do this w/ the configuration settings, but we couldn't enforce the mentor mode options in that only one option could be selected):
- Enable mentor mode for all judges
- Enable mentor mode only for judges who already have a mentor profile
- Off
- Enable judge mode for all mentors
- Off

**Comment by shaun-technovation** [comment]
Based on the convo in standup, we will add three new config settings:

- `ENABLE_MENTOR_MODE_FOR_ALL_JUDGES`
- `ENABLE_MENTOR_MODE_ONLY_FOR_JUDGES_WITH_EXISTING_JUDGE_PROFILE`
- `ENABLE_JUDGE_MODE_FOR_ALL_MENTORS`

Only one `MENTOR_MODE` option should be set at any given time, we'll have to remember this since we'll be manually adding/editing these settings. When we convert these setting to admin settings (#4679) we can enforce it (via the GUI) so that only one option can be selected.

**Comment by shaun-technovation** [comment]
These new settings are available on QA now:

- `ENABLE_MENTOR_MODE_FOR_ALL_JUDGES`
- `ENABLE_MENTOR_MODE_ONLY_FOR_JUDGES_WITH_EXISTING_JUDGE_PROFILE`
- `ENABLE_JUDGE_MODE_FOR_ALL_MENTORS`

Currently, these settings are enabled on QA:
- `ENABLE_MENTOR_MODE_ONLY_FOR_JUDGES_WITH_EXISTING_JUDGE_PROFILE`
- `ENABLE_JUDGE_MODE_FOR_ALL_MENTORS`

And FYI, judge/mentor mode switching doesn't work when impersonating, but when logging in as the actual mentor or judge it works.

**Comment by dboyer** [comment]
Tested on QA with these two parameters (ENABLE_MENTOR_MODE_ONLY_FOR_JUDGES_WITH_EXISTING_JUDGE_PROFILE and ENABLE_JUDGE_MODE_FOR_ALL_MENTORS) and confirmed:

- As a judge with no mentor account, I do not see the mentor mode button
- As a mentor, I saw a link to the judge dashboard
- As that mentor on the judge dashboard, I had a link back to mentor mode

I then changed the settings to be ENABLE_MENTOR_MODE_FOR_ALL_JUDGES and ENABLE_JUDGE_MODE_FOR_ALL_MENTORs and confirmed:

- As a mentor, I had a link to the judge dashboard on my mentor dashboard
- As that same mentor, I had a link on my judge dashboard to go back to the mentor profile
- As a judge with no mentor profile, I had the link to mentor mode

I then deleted the ENABLE_JUDGE_MODE_FOR_ALL_MENTOR variable and confirmed:

- As a judge who then went to mentor dashboard, I did not have the link to the judge dashboard
- As a mentor with a previously existing judge account, I also did not have a link to the judge dashboard from my mentor dashboard

I confirmed the mentor mode button appears irrespective of the mentor registration setting, which seems correct.

@shaun-technovation One wording note, should it be ENABLE_MENTOR_MODE_ONLY_FOR_JUDGES_WITH_EXISTING_MENTOR_PROFILE? Right now, it's WITH_EXISTING_JUDGE_PROFILE.
